### ISSUE TITLE
Reveal secrets and replace refs in place

### ISSUE THREAD

**Issue by smauermann** [kapicorp/kapitan#827]
**Describe the bug/feature**
We would like to compile our templates without revealing secrets, i.e. without the `--reveal` flag. Later on we would like to consume the compiled assets and reveal the secrets in place, i.e. replace the secret refs in the compiled files with the actual secret values.

**Expected behavior**
1. Compile templates via `kapitan compile --target ${TARGET}` => Compiled files contain secret refs
2. Reveal secrets in place via, e.g. with a new flag `kapitan refs --reveal --inplace` => Compiled files contain actual secret values

**Additional context**
This would be useful to be able to safely share compiled templates between build jobs via artifacts.

**Comment by smauermann** [comment]
@ramaro thanks for getting back to me. The `--embed-refs` option is not working for us as we want to achieve:
- Compiled Terraform and Kubernetes specs that only contain secret refs
- Update the specs in place and convert secret refs to their actual values (in place as opposed to displaying the revealed specs in STDOUT)

Does that make sense to you? Or is it already possible and I am missing something?

We could use this in our CI/CD pipelines, where we have an initial `kapitan compile` step which generates the resource specs. The compiled specs are shared between CI jobs via artifacts (GitLab). The problem here is that the artifacts contain sensitive values that are easily accessible. Ideally we want only refs to the secrets and not plaintext secrets in our artifacts. The alternative is to run `kapitan compile` in each CI job and not share compiled specs via artifacts. Performing the same task over and over again seems redundant though.

**Comment by ramaro** [comment]
Oh, I misunderstood you. Right now you can run `kapitan compile --reveal` at the stage where it makes sense to have revealed artefacts. Let me know if that works for you!

**Comment by ramaro** [comment]
Yes that is a valid point. Compiling again is not ideal and the ref reveal command could offer such functionality, but it's a decent workaround.
I will leave this open to track the issue to be worked on. thanks

**Comment by ademariag** [comment]
Late to the party. @smauermann do you mind joining the #kapitan channel on the kubernetes slack to discuss?

Questions:
* what secret backend are you using? gkms?
* @smauermann for single file reveal, could you not use `kapitan refs --reveal -f file-to-reveal`
* What needs to access the secrets from within the CI? Terraform?
